### PR TITLE
Mpdx 6975 add mutation for updating contact tags

### PR DESCRIPTION
**Pull request by gyasistory**
This PR is to add delete functionality to Contact Chip.  I used the mutation to update the chip

### CODE REVIEW

**Review by reldredge71 — APPROVED** [review]

File: src/components/Contacts/ContactDetails/ContactDetailsTab/Tags/ContactTags.tsx [hunk 1]
```diff
@@ -21,22 +25,37 @@ const ContactTagAddTagText = styled(Typography)(({ theme }) => ({
...
 export const ContactTags: React.FC<ContactTagsProps> = ({
+  accountListId,
   contactId,
-  contactTags,
 }) => {
   const { t } = useTranslation();
+  const { data, loading } = useGetContactTagsQuery({
+    variables: {
+      accountListID: accountListId, // value for 'accountListID'
+      contactId: contactId, // value for 'contactId'
```
Comment (OzzieOrca) [inline src/components/Contacts/ContactDetails/ContactDetailsTab/Tags/ContactTags.tsx:40]: I don't think these comments add any value.

File: src/components/Contacts/ContactDetails/ContactDetailsTab/Tags/ContactTags.tsx [hunk 2]
```diff
@@ -21,22 +25,37 @@ const ContactTagAddTagText = styled(Typography)(({ theme }) => ({
...
+  });
+
   const handleTagDelete = (tag: string) => {
-    const index = contactTags.indexOf(tag);
-    if (index > -1) {
-      contactTags = contactTags.splice(index, 1);
-      // TODO: add mutation to update tag for contact
-    }
-    console.info('You clicked the Delete tag for tag' + tag + '' + contactId);
+    console.info('You clicked the Delete tag for tag ' + tag + ' ' + contactId);
```
Comment (OzzieOrca) [inline src/components/Contacts/ContactDetails/ContactDetailsTab/Tags/ContactTags.tsx:45]: Probably don't need to keep this.

File: src/components/Contacts/ContactDetails/ContactDetailsTab/Tags/ContactTags.tsx [hunk 3]
```diff
@@ -21,22 +25,37 @@ const ContactTagAddTagText = styled(Typography)(({ theme }) => ({
...
+    const list = data.contact.tagList;
+    const updatedTags: string[] = [];
+    list.forEach((element) => {
+      element != tag ? updatedTags.push(tag) : null;
+    });
```
Comment (OzzieOrca) [inline src/components/Contacts/ContactDetails/ContactDetailsTab/Tags/ContactTags.tsx:50]: `.filter` would be better here. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

File: src/components/Contacts/ContactDetails/ContactDetailsTab/Tags/ContactTags.tsx [hunk 4]
```diff
@@ -21,22 +25,37 @@ const ContactTagAddTagText = styled(Typography)(({ theme }) => ({
...
+
+    const [] = useUpdateContactTagsMutation({
+      variables: {
+        accountList: accountListId,
+        contactId: contactId,
+        tagList: updatedTags,
+      },
+    });
```
Comment (OzzieOrca) [inline src/components/Contacts/ContactDetails/ContactDetailsTab/Tags/ContactTags.tsx:58]: Hooks need to be at the top level of a react component. You can't put them inside other functions. This hook returns a function that you can use to call it when desired.
I would use the hook like this:
```ts
const [updateTags, { data }] = useUpdateContactTagsMutation({
  variables: {
    accountList: accountListId,
    contactId: contactId,
  },
});
```
and then later call the `updateTags` function to execute the mutation when needed
```ts
updateTags({
  variables: {
    tagList: updatedTags,
  },
});
```
You can set variables either in the hook or when you call the mutation. In this case you don't know the tags until you're inside this function so you can wait to set that until it's executed.

https://www.apollographql.com/docs/react/data/mutations/#executing-a-mutation

File: src/components/Contacts/ContactDetails/ContactDetailsTab/Tags/ContactTags.tsx [hunk 5]
```diff
@@ -21,22 +25,37 @@ const ContactTagAddTagText = styled(Typography)(({ theme }) => ({
...
-  contactTags: string[];
 }
 
 export const ContactTags: React.FC<ContactTagsProps> = ({
+  accountListId,
   contactId,
-  contactTags,
 }) => {
   const { t } = useTranslation();
+  const { data, loading } = useGetContactTagsQuery({
```
Comment (OzzieOrca) [inline src/components/Contacts/ContactDetails/ContactDetailsTab/Tags/ContactTags.tsx:37]: Could the parent component load these? You could pass it a fragment.

**Review by OzzieOrca — CHANGES_REQUESTED** [review]

**Pull request closed without merging**